lib/Graph/Graph.cpp

```diff
@@ -1923,6 +1923,34 @@ void Function::createLSTM(llvm::StringRef namePrefix,
 //                   Placeholder-builder methods.
 //===----------------------------------------------------------------------===//
 
+BatchNormalizationNode *
+Function::createBatchNormalization(Context &ctx, llvm::StringRef name, NodeValue input,
+                                   unsigned_t channelIdx, float epsilon,
+                                   float momentum) {
+  // Figure out how many channels are in the tensor.
+  size_t channels = input.dims()[channelIdx];
+
+  // Allocate the learnable parameters beta and gamma.
+  auto *beta = getParent()->createPlaceholder(
+      ElemKind::FloatTy, {channels}, "beta", true);
+  ctx.allocate(beta)->init(glow::Tensor::InitKind::Zero, 0, getPRNG());
+
+  auto *gamma = getParent()->createPlaceholder(
+      ElemKind::FloatTy, {channels}, "gamma", true);
```

Comment:
curious, why this is always trainable?

Reply:
Hm, I was just copying this from the existing (non-Context) createBatchNorm.  Should it be otherwise?

Reply:
I think that this is correct. Instead of creating trainable variables we create trainable placeholders and initialize them.